source/state_representation/test/tests/space/cartesian/test_cartesian_pose.cpp

```diff
@@ -0,0 +1,277 @@
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_wrench().norm(), 0);
+  // try to change non pose variables prior to the copy, those should be discarded
+  static_cast<CartesianState&>(pose1).set_twist(Eigen::VectorXd::Random(6));
+  static_cast<CartesianState&>(pose1).set_accelerations(Eigen::VectorXd::Random(6));
+  static_cast<CartesianState&>(pose1).set_wrench(Eigen::VectorXd::Random(6));
+  CartesianPose pose4 = pose1;
+  EXPECT_TRUE(pose1.data().isApprox(pose4.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_wrench().norm(), 0);
+  // copy a state, only the pose variables should be non 0
+  CartesianPose pose5 = CartesianState::Random("test");
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_wrench().norm(), 0);
+
+  CartesianPose pose6;
+  EXPECT_TRUE(pose6.is_empty());
+  CartesianPose pose7 = pose6;
+  EXPECT_TRUE(pose7.is_empty());
+}
+
+TEST(CartesianPoseTest, SetData) {
+  CartesianPose cp1 = CartesianPose::Identity("test");
+  CartesianPose cp2 = CartesianPose::Random("test");
+  cp1.set_data(cp2.data());
+  EXPECT_TRUE(cp2.data().isApprox(cp1.data()));
+
+  auto pose_vec = cp2.to_std_vector();
+  cp1.set_data(pose_vec);
+  for (std::size_t j = 0; j < pose_vec.size(); ++j) {
+    EXPECT_FLOAT_EQ(pose_vec.at(j), cp1.data()(j));
+  }
+}
+
+TEST(CartesianPoseTest, CartesianPoseToStdVector) {
+  CartesianPose cp = CartesianPose::Random("test");
+  std::vector<double> vec_data = cp.to_std_vector();
+  EXPECT_EQ(vec_data.size(), 7);
+  for (size_t i = 0; i < vec_data.size(); ++i) {
+    EXPECT_EQ(cp.data()(i), vec_data[i]);
+  }
+}
+
+TEST(CartesianPoseTest, MultiplyTransformsBothOperators) {
+  Eigen::Vector3d pos1(1, 2, 3);
+  Eigen::Quaterniond rot1(1, 0, 0, 0);
+  CartesianPose tf1("t1", pos1, rot1);
+  Eigen::Vector3d pos2(4, 5, 6);
+  Eigen::Quaterniond rot2(1, 0, 0, 0);
+  CartesianPose tf2("t2", pos2, rot2, "t1");
+  CartesianPose tf3 = tf1 * tf2;
+  tf1 *= tf2;
+  EXPECT_EQ(tf3.get_name(), "t2");
+  for (int i = 0; i < tf1.get_position().size(); ++i)
+    EXPECT_NEAR(tf1.get_position()(i), tf3.get_position()(i), 0.00001);
+}
+
+TEST(CartesianPoseTest, MultiplyTransformsSameOrientation) {
+  Eigen::Vector3d pos1(1, 2, 3);
+  Eigen::Quaterniond rot1(1, 0, 0, 0);
+  CartesianPose tf1("t1", pos1, rot1);
+  Eigen::Vector3d pos2(4, 5, 6);
+  Eigen::Quaterniond rot2(1, 0, 0, 0);
+  CartesianPose tf2("t2", pos2, rot2, "t1");
+  tf1 *= tf2;
+  Eigen::Vector3d pos_truth(5, 7, 9);
+  for (int i = 0; i < pos_truth.size(); ++i) {
+    EXPECT_NEAR(tf1.get_position()(i), pos_truth(i), 0.00001);
+  }
+}
```

Comment:
This test case can be removed and combined with the previous test case, just by adding the expected position.